optima/calibration.py

```diff
@@ -224,14 +243,79 @@ def extractdata(xdata, ydata):
                         if nrows==1: modelrow = modelrows # WARNING, kludgy, should have the same shape!
                         else: modelrow = modelrows[row]
                         datax, datay = extractdata(results.datayears, datarow) # Pull out the not-NaN values
+                        if doplot: rowname = 'total' if nrows==1 else pars['popkeys'][row]
                         for i,year in enumerate(datax): # Loop over each data point available
+                            count += 1
                             modelx = findinds(results.tvec, year) # Find the index of the corresponding time point
                             modely = modelrow[modelx] # Finally, extract the model result!
-                            thismismatch = abs(modely - datay[i]) / mean(datay+eps)
+                            if method=='mad': thismismatch = abs(modely - datay[i]) / mean(datay+eps)
```

Comment:
this isn't really the MAD, though - MAD would not normalise. how about weighted absolute percent error?